src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt

```diff
@@ -54,33 +57,63 @@ abstract class PluginRestTestCase : OpenSearchRestTestCase() {
     }
 
     open fun preserveOpenSearchIndicesAfterTest(): Boolean = false
+    open fun preserveOpenSearchDataStreamsAfterTest(): Boolean = true
 
     @Throws(IOException::class)
     @After
     open fun wipeAllOpenSearchIndices() {
-        if (preserveOpenSearchIndicesAfterTest()) return
-        val response = client().performRequest(Request("GET", "/_cat/indices?format=json&expand_wildcards=all"))
-        val xContentType = XContentType.fromMediaType(response.entity.contentType)
-        xContentType.xContent().createParser(
-            NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
-            response.entity.content
-        ).use { parser ->
-            for (index in parser.list()) {
-                val jsonObject: Map<*, *> = index as java.util.HashMap<*, *>
-                val indexName: String = jsonObject["index"] as String
-                // .opendistro_security isn't allowed to delete from cluster
-                if (".opendistro_security" != indexName) {
-                    val request = Request("DELETE", "/$indexName")
-                    // TODO: remove PERMISSIVE option after moving system index access to REST API call
-                    val options = RequestOptions.DEFAULT.toBuilder()
-                    options.setWarningsHandler(WarningsHandler.PERMISSIVE)
-                    request.options = options.build()
-                    adminClient().performRequest(request)
+        if (!preserveOpenSearchDataStreamsAfterTest()) {
+            var getResponse = executeRequest(
+                RestRequest.Method.GET.name,
+                "/_data_stream/*",
+                "",
+                RestStatus.OK.status
+            )
+            Assert.assertNotNull(!getResponse.get("data_streams").asJsonArray.isEmpty)
+            for (stream in getResponse.get("data_streams").asJsonArray) {
+                dataStreamCleanUp(stream.asJsonObject.get("name").asString)
+            }
+        }
+        if (!preserveOpenSearchIndicesAfterTest()) {
+            val response = client().performRequest(Request("GET", "/_cat/indices?format=json&expand_wildcards=all"))
+            val xContentType = XContentType.fromMediaType(response.entity.contentType)
+            xContentType.xContent().createParser(
+                NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                response.entity.content
+            ).use { parser ->
+                for (index in parser.list()) {
+                    val jsonObject: Map<*, *> = index as java.util.HashMap<*, *>
+                    val indexName: String = jsonObject["index"] as String
+                    // .opendistro_security isn't allowed to delete from cluster and same for data-stream backing indices
+                    if (".opendistro_security" != indexName && !indexName.startsWith(".ds-", false)) {
```

Comment:
how are .ds-* indices created? would they be system indices?

Reply:
I'll remove this section for now - the new  test is not expecting the default index to be created only the template